sphinx/registry.py

```diff
@@ -21,7 +21,11 @@
 from docutils.parsers import Parser
 from docutils.parsers.rst import Directive
 from docutils.transforms import Transform
-from pkg_resources import iter_entry_points
+
+try:  # Python < 3.10 (backport)
```

Comment:
FYI the try/except idiom is known to confuse mypy. There's a known bug that hasn't been fixed for years AFAIK. The workaround is to use if/else.

Reply:
mypy 0.921 seems to be happy with this - as far as I can tell, the mypy CI failures are due to the recent release of mypy 0.930, and entirely orthogonal to this PR.

Reply:
Yeah, looks like you're right.